Bugzilla/DuoClient.pm

```diff
@@ -0,0 +1,264 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+#
+# Adapted from https://github.com/duosecurity/duo_universal_python/blob/main/duo_universal/client.py
+
+package Bugzilla::DuoClient;
+
+use 5.10.1;
+use Moo;
+
+use Bugzilla::Constants;
+use Bugzilla::Error;
+use Bugzilla::Logging;
+use Bugzilla::Util qw(generate_random_password mojo_user_agent);
+
+use Mojo::JWT;
+use Mojo::URL;
+use Try::Tiny;
+use Types::Standard -types;
+
+use constant JTI_LENGTH             => 36;
+use constant MINIMUM_STATE_LENGTH   => 22;
+use constant MAXIMUM_STATE_LENGTH   => 1024;
+use constant JWT_DEFAULT_EXPIRATION => 300;
```

Comment:
Nit: should probably add `_SECONDS` so it's obvious what the units are here. I realize that this name was my own suggestion, sorry about that. :sweat_smile: